### DISCUSSION TITLE
Can you disable form submissions on use of the 'enter' key on an input field globally?

### DISCUSSION THREAD

**Discussion by vpoile** [Laravel-Backpack/community-forum#412] [General] [answered]
Is it possible to globally prevent the enter key from submitting the CRUD create/edit panels. My users are getting confused between the rich text areas that allow the enter key to be submitted and regular input fields which lead them to making mistakes.

Apologies if this is already answered I have looked in both the documentation and on this forum but can't see an answer, don't want to add functionality if its already possible within the existing build.

**Comment by o15a3d4l11s2** [comment]
You could do something like
```
$("form").bind("keypress", function(e) {
    if (e.keyCode == 13) {
        return false;
    }
});
```
or similar to "catch" the ENTER key and disable it from triggering the form. For this to work you would need to add it as for any/all forms you desire. The problem with this solution is that it will disable ENTER also on your rich text fields. So you should actually target not the form itself, but the inputs.

Another possibility would be to do a "fake" submit field. This would work well in custom forms, but not so easy in the CRUD.

So my personal suggestion would be to use the first example with a selector like `$("form input[type=text]")` and fine-tune it for the specific fields that you want to target. As a more straightforward solution, but with more manual work, you could add a special class like `no-submit`and target only these fields in your selector ``$("form .no-submit")`.

Sorry for the long post, just giving different ideas so you can decide what would work for your specific needs. To put it simple: there is no default easy way to disable form submit on ENTER.

**Reply by tabacitu** [reply, marked as answer]
Yup. I'd only add one thing - that whatever JS you add to the page, you can do it using [the script widget](https://backpackforlaravel.com/docs/5.x/base-widgets#script-1).
